speculation-rules.bs

```diff
@@ -649,6 +649,27 @@ The same mitigations available to other features also apply here. In particular,
 
 The possibility of leaking link URLs via this mechanism is additionally mitigated by the fact that prefetch and prerender to plaintext HTTP (other than to localhost) is not permitted, and so such an on-path attacker could not directly observe preloading request URLs, but would only have access to metadata and traffic analysis. This does not, however, replace standard XSS protections.
 
+<h3 id="security-inline-scripting">Inline scripting</h3>
+
+Inline script use is also restricted by the [[CSP]], but there is a case a developer doesn't want to permit general inline scripts, but want to permit inline speculation rules. So, we make the following change that extends [[CSP]] to support a new source keyword for inline speculation rules.
+
+In the [[CSP#framework-directive-source-list]], we change the following definition to have a new source, `inline-speculation-rules`.
+
+<pre dfn-type="grammar" link-type="grammar">
+  <dfn>keyword-source</dfn> = "<dfn>'self'</dfn>" / "<dfn>'unsafe-inline'</dfn>" / "<dfn>'unsafe-eval'</dfn>"
+    / "<dfn>'strict-dynamic'</dfn>" / "<dfn>'unsafe-hashes'</dfn>"
+    / "<dfn>'report-sample'</dfn>" / "<dfn>'unsafe-allow-redirects'</dfn>"
+    / "<dfn>'wasm-unsafe-eval'</dfn>" / "<dfn>'inline-speculation-rules'</dfn>"
+</pre>
+
+Also, we make a following change for a note in [[CSP#should-block-inline]].
+
+Note: The valid values for `type` are "`script`", "`script attribute`", "`script speculationrules`", "`style`", and "`style attribute`".
```

Comment:
Does anything actually supply the "script speculationrules" value right now? Do we need an additional patch to plumb that through?

Reply:
https://w3c.github.io/webappsec-csp/#ref-for-should-block-inline
There are two references, and probably I should modify this description.
Currently, we don't have a clear statement how each value is chosen. Let me try adding it.